storefront/app/views/spree/checkout/_payment_methods.html.erb

```diff
@@ -42,15 +42,19 @@
 
             <% if selected_method == method %>
               <div class="px-5 py-4">
-                <%= render partial: "spree/checkout/payment/#{method.method_type}", locals: { payment_method: method } %>
+                <%= render_with_default partial: "spree/checkout/payment/#{method.type.demodulize.downcase}", locals: { payment_method: method } do %>
+                  <%= render partial: "spree/checkout/payment/#{method.method_type}", locals: { payment_method: method } %>
+                <% end %>
               </div>
             <% end %>
           </li>
         <% end %>
       </ul>
     <% elsif checkout_available_payment_methods.any? %>
       <%= hidden_field_tag "order[payments_attributes][][payment_method_id]", checkout_available_payment_methods.first&.id %>
-      <%= render partial: "spree/checkout/payment/#{checkout_available_payment_methods.first.method_type}", locals: { payment_method: checkout_available_payment_methods.first } %>
+      <%= render_with_default partial: "spree/checkout/payment/#{checkout_available_payment_methods.first.type.demodulize.downcase}", locals: { payment_method: checkout_available_payment_methods.first } do %>
```

Comment:
@caiodsc Could you explain what's the added value here? We already allow using custom forms, we can define the partial name through `method_type`. 

By default, it's always `gateway` which we can maybe remove? Then we'll always use the `Spree::Payment#method_type` which is `type.demodulize.downcase` - the same thing you're doing here with `render_with_default`.